mobile/src/screens/BookScreen.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { Text, View, StyleSheet, Image, ScrollView } from 'react-native';
+import { useRoute } from '@react-navigation/native';
+import { MarkdownView } from 'react-native-markdown-view';
+import YoutubePlayer from 'react-native-youtube-iframe';
+import { BookScreenRouteProps } from '../navigation/HomeStackNavigator';
+import { LoadingCircle } from '../components/LoadingCircle';
+import { APIContext } from '../context/APIContext';
+import { I18nContext } from '../context/I18nContext';
+import { TextStyles } from '../styles/TextStyles';
+import { ButtonGroup } from '../components/ButtonGroup';
+import { LanguageButtons } from '../components/LanguageButtons';
+import { Colors } from '../styles/Colors';
+
+/**
+ * Individual book view displaying book details
+ */
+export const BookScreen: React.FC = () => {
+  // get book id and langs from route params
+  const route = useRoute<BookScreenRouteProps>();
+  const { book } = route.params;
+  const langs = book.languages;
+
+  const client = useContext(APIContext);
+  const i18nCtx = useContext(I18nContext);
+
+  const locale = i18nCtx.locale.substring(0, 2);
+  const defaultLang = langs.includes(locale) ? locale : langs.includes('en') ? 'en' : langs[0];
+
+  const videoIdRegEx = '^(?:https?:)?//[^/]*(?:youtube(?:-nocookie)?.com|youtu.be).*[=/]([-\\w]{11})(?:\\?|=|&|$)';
+
+  // book screen states
+  const [loading, setLoading] = useState(true);
+  const [bookDetails, setBookDetails] = useState(null);
+  const [language, setLanguage] = useState(defaultLang);
+  const [activeButton, setActiveButton] = useState('read');
```

Comment:
Can you type these for a little extra safety? There are a lots of `any`s being thrown around here and I'm surprised TS hasn't been reporting some of them as errors...

You don't need to type `loading` since that's obviously a bool (and TS knows that), but the rest of these need to be more explicit:

```ts
  const [bookDetails, setBookDetails] = useState<BookDetails>(null);
  const [language, setLanguage] = useState<Language>(defaultLang);
  const [activeButton, setActiveButton] = useState<Tab>('read');
```

For the Tab type, you can define that somewhere at the top of this file as `type Tab = 'read' | 'explore' | 'learn'`